B/crud supplier

juro95 opened the pull request:
hab jetzt mal den rest von crud für supplier gemacht
müssten noch tests schreiben, hab jetzt erstmal manuell getestet. hab es jetzt so das wir einfach alles vom supplier löschen auf delete, aber das können wir ja sonst noch ändern auch

juro95 committed:
supplier patch works

juro95 committed:
delete supplier

julklingel reviewed (commented):
Top, bin aber am überlegen ob wir nicht den Datei-Upload und Adressaktualisierung in getrennte Funktionen packen, um die patch Funktion übersichtlicher zu gestalten.
Können wir aber noch in einem Refaktor Schritt später machen. :)

julklingel commented:
Oh und es fehlen noch checks. Man kann beispielsweise companyName leer lassen. Hier brauchen wir @isnotempty decorator im dto. Außerdem wenn sich der Firmen Name ändert müsste sich auch der Slug ändern.

juro95 commented:
ich schau mal das ich es nochwas refactor :) 
bei dem updateAddressDto und  updateSupplierDto hab ich isOptional überall gemacht damit man einfach nur das hinschicken kann was man updaten möchte und nicht alles schicken muss oder?

juro95 commented:
und stimmt an den slug hatte ich nicht gedacht!

juro95 committed:
refactor get methods

julklingel commented:
> Author

Jo passt, aber wir müssen trotzdem verhindern das sie einen leeren String einreichen.

juro95 committed:
refactored image upload and fixed bug

habe gerade gemerkt das man das image nur updaten kann wenn die column nicht leer ist also muss man erst checken ob ein image schon da ist

juro95 committed:
delete logs

juro95 committed:
delete image from cloudinary on catch

juro95 committed:
setting min length to 1

juro95 commented:
ok ist jetzt alles drin und funktioniert
hab auch noch hinzugefügt dass das image auf cloudinary gelöscht wird wenn was beim hochladen in die datenbank schifläuft
